test/validate_test.js

```diff
@@ -112,63 +156,79 @@ describe("Validation", () => {
           [illFormedKey]: { type: "string" },
         },
       };
-      const newErrorMessage = "Better error message";
-      const transformErrors = errors => {
-        return [Object.assign({}, errors[0], { message: newErrorMessage })];
-      };
-
-      let errors;
 
-      beforeEach(() => {
+      it("should use transformErrors function", () => {
+        let errors;
+        const newErrorMessage = "Better error message";
+        const transformErrors = errors => {
+          return [Object.assign({}, errors[0], { message: newErrorMessage })];
+        };
         const result = validateFormData(
           { foo: 42, [illFormedKey]: 41 },
           schema,
           undefined,
           transformErrors
         );
         errors = result.errors;
-      });
 
-      it("should use transformErrors function", () => {
         expect(errors).not.to.be.empty;
         expect(errors[0].message).to.equal(newErrorMessage);
       });
+
+      it("call after custom validations", () => {
+        let errors;
+        const transformErrors = errors => {
+          return errors.sort(a => (a.name === "custom" ? -1 : 1));
+        };
+        const validate = (formData, errors) => {
+          errors.foo.addError("custom");
+          return errors;
+        };
+        const result = validateFormData(
+          { foo: 42, [illFormedKey]: 41 },
+          schema,
+          validate,
+          transformErrors
+        );
+        errors = result.errors;
+
+        expect(errors).not.to.be.empty;
+        expect(errors[0].stack).to.equal("foo: custom");
```

Comment:
Can you add an assertion to make sure that the `errors[0].name` is also equal to `custom`?

Reply:
That's actually not available here because the schemas are [converted to an errorList](https://github.com/mozilla-services/react-jsonschema-form/blob/18503088e57aaf11af0cc64d7a70d0e330c1127a/src/validate.js#L234-L239) whose items only have a `stack` prop for the return value of  `validateFormData`

`transformErrors` functions are given the flattened data that includes the `name` and here on L181 I sort by that. The `stack` assertion validates that the sorting worked. An AJV `foo: should be string` error comes first without sorting.